### DISCUSSION TITLE
Basis of Fundamental Physical Constants?

### DISCUSSION THREAD

**Discussion by nerffan1** [FLOSIC/PublicRelease_2020#32] [General]
Hello,

I am trying to write a section for physical constants used in FLOSIC for the _new_ [documentation](https://github.com/nerffan1/FLOSICDOC). Are we using [CODATA](https://www.nist.gov/programs-projects/codata-values-fundamental-physical-constants) values or something else?

Thank You!

**Comment by peraltajuan** [comment]
Hi Angel,
As far as I know, the FLOSIC code does not use any standardized physical constants data. Some other codes load all this data in either common blocks or modules to reuse across the code. Is that something that could be implemented?




Juan E. Peralta
Professor, Department of Physics
Director,  Science of Advanced Materials Program
Central Michigan University
http://people.cst.cmich.edu/peral1j/

From: Angel-Emilio Villegas Sanchez ***@***.***>
Date: Wednesday, July 12, 2023 at 2:12 PM
To: FLOSIC/PublicRelease_2020 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [External] [FLOSIC/PublicRelease_2020] Basis of Fundamental Physical Constants? (Discussion #32)

Hello,

I am trying to write a section for physical constants used in FLOSIC for the new documentation<https://github.com/nerffan1/FLOSICDOC>. Are we using CODATA<https://www.nist.gov/programs-projects/codata-values-fundamental-physical-constants> values or something else?

Thank You!

**Reply by nerffan1** [reply]
It might be simple(?)This [repo](https://github.com/vmagnin/fundamental_constants) looks useful for an implementation. The hardest part sounds like making a **complete** list of all constants that are being used. We could just 
1. `grep` for the values in the source code
2. Make sure the module is accessible to those files
3. Replace old constants with module constants using `sed`.

**Comment by yohyamamoto** [comment]
The constants used in the code are mainly for unit conversions or parameters in XC, vDW, etc.

```
DATA HA2EV/27.2116D0/
DATA HA2KEL/315891.1826D0/
au2ang=0.52917721067d0 !CODATA 2014 !0.529177
ang2au=1.0d0/au2ang    !1.889726878
autokcal=627.509541d0
autoev  = 27.21138505
DATA CLIGHT/137.088D0/
DATA FACT/0.148203857088/
PI=4.0D0*DATAN(1.0D0)
DATA DKF2RS/1.9191582926775128D0/
```

And atom mass data:
```
  DATA ATMASS/0.0, 1.00797, 4.0026, 6.939, 9.0122, 10.811, 12.01115
&            , 14.0067, 15.9994, 18.9984, 20.179, 22.9898, 24.305
&            , 26.9815, 28.086, 30.9738, 32.064, 35.453, 39.948
&            , 39.102, 40.08, 44.956, 47.90, 50.942, 51.996
&            , 54.9381, 55.847, 58.9332, 58.71, 63.546, 65.37
&            , 69.72, 72.59, 74.9216, 78.96, 79.904, 83.80, 85.47
&            , 87.62, 88.905, 91.22, 92.906, 95.94, 98.906, 101.07
&            , 102.905, 106.4, 107.868, 112.40, 114.82, 118.69
&            , 121.75, 127.60, 126.9044, 131.30, 132.905, 137.34/
```

**Reply by nerffan1** [reply]
I found several cases where the constants are declared/initialised, but they are not used in the rest of the subroutine, e.g. : 

```
$grep -ni AU2ANG moss1.f 
22:       REAL*8 :: SYMBOL , RHOFERMI, EFGRAD, CENTER, ANGLE, AU2ANG,
56:       DATA AU2ANG/0.529177D0/
```
and 
```
$grep -ni HA2EV kinm3d.f 
14:       DATA HA2EV/27.2116D0/
```

These can be deleted correct?

**Reply by yohyamamoto** [reply]
That is correct, and moss1 is not in use.
